### DISCUSSION TITLE
Basic questions on Syncing, Microsoft 365 apps, and uninstalling

### DISCUSSION THREAD

**Discussion by annatomaselli** [abraunegg/onedrive#2645] [Q&A] [answered]
Hi there,

I am **very very new** to Ubuntu, I just managed to sync locally my OneDrive files using the instructions but... this is not actually what I was looking for and I'm struggling to understand what to do next:
I thought that this "client" (hope that's the right term) would allow me to _access_ my OneDrive files from the cloud _without_ downloading them locally and having to resync via the terminal everytime. I went through past questions on this forum but I am still very confused, so just to confirm:

1) This client allows me to **access files _locally_ and then _manually_ sync** them to the online OneDrive folder, but it does not allow 'on-demand' file access (?): so just to be clear, this is **not** possible in a straightforward way like in Windows, right? 
Another user asked for this [here](https://github.com/abraunegg/onedrive/discussions/2630) but I really don't understand what a Docker is unfortunately, so I don't understand the answer either...

2) Now that the files have been downloaded locally, when I modify them or delete them I am actually only interacting with the local copy and not with the original one in the cloud, right? I can then sync whatever I modified locally either by `onedrive --synchronize --local-first` or by `onedrive --synchronize --upload-only` , and I think I read something about automatic syncs somewhere. For example [here](https://github.com/abraunegg/onedrive/discussions/2391) it is mentioned that when using `--synchronize` then the files are only synced when "running the application"... I'm really sorry but which application are we referring to? Since OneDrive is not actually installed as an app as far as I understand...

3) I thought that I should be able to install other Microsoft 365 apps (outlook, word, etc) through this "client", but I can't figure out where the instructions are for this. Sorry if that's not the case at all.

4) Because I use OneDrive to avoid having to download all my files locally, I'm afraid this "client" might not work for me. I have Ubuntu 22.04 / Jammy so I installed it following [these](https://github.com/abraunegg/onedrive/blob/master/docs/ubuntu-package-install.md#distribution-ubuntu-2204) instructions. To uninstall them, following what's written [here](https://github.com/abraunegg/onedrive/blob/master/docs/INSTALL.md#uninstalling-the-client), it says to do `sudo make uninstall` from "within your GitHub repository clone", but.... I do not have a GitHub repository clone? I just installed this through my terminal and I'm not sure how to uninstall what I have done without possibly messing up my OneDrive folders and files (which I do not have backed up anywhere else due to the size...).

So just to confirm, **if I were to manually delete all the downloaded OneDrive folders locally _without_ syncing on the terminal afterwards, that would not affect my original files on the cloud, am I correct??** 
And if I were to (understand and) follow the uninstalling instructions mentioned above, would that get rid of all the downloaded files from my machine but not from the OneDrive cloud?

I'm very sorry for the basic questions but I figured that this could be useful for future users who are also beginners like me...

Thank you very much!

**Comment by abraunegg** [comment, marked as answer]
@annatomaselli 
> I am very very new to Ubuntu

Whilst this is 100% your choice, I would potentially suggest then look at different distributions to avoid yourself lots of future pain where *other* packages are broken with bugs and issues but are not fixed because of their policy to use older packages in the guise of stability.

> 1. This client allows me to access files locally and then manually sync them to the online OneDrive folder, but it does not allow 'on-demand' file access (?): so just to be clear, this is not possible in a straightforward way like in Windows, right?

This client does *not* provide on-demand access. This is called out in the [readme](https://github.com/abraunegg/onedrive/blob/master/README.md) file:
![image](https://github.com/abraunegg/onedrive/assets/4956234/abaebfe1-3ec0-4e0f-83ed-6dad26e5e059)

To use this capability, you *must* use 'onedriver' which can be found here: https://github.com/jstaf/onedriver

Both applications cannot co-exist.

To answer the 'manual' sync part - there are 2 operational modes for the client:
* Manual sync - this involves you running the client using `onedrive --synchronize`
* Automatic sync - this involves you running the client using `onedrive --monitor` so that it monitors your filesystem for changes. You can also set up a systemd service so that it monitors your system for changes in the background.

> 2. Now that the files have been downloaded locally, when I modify them or delete them I am actually only interacting with the local copy and not with the original one in the cloud, right?

Correct - you are only modifying your local copy of the file. You can then sync your changes back to OneDrive online using one of the methods above.

> ... Since OneDrive is not actually installed as an app as far as I understand...

Any executable binary you install is an application .. or app .. so when you installed the client, you installed 'onedrive' an executable binary which is an application (or app)

> 3. I thought that I should be able to install other Microsoft 365 apps (outlook, word, etc) through this "client", but I can't figure out where the instructions are for this. Sorry if that's not the case at all.

This has nothing to do with the 'onedrive' client provided here. All the Microsoft 365 applications will be available using the online version via Chrome. If you want to install an OpenOffice Suite:
* LibreOffice
* OpenOffice

Stay away from WPS Office for many reasons

> 4. Because I use OneDrive to avoid having to download all my files locally, I'm afraid this "client" might not work for me. I have Ubuntu 22.04 / Jammy so I installed it following [these](https://github.com/abraunegg/onedrive/blob/master/docs/ubuntu-package-install.md#distribution-ubuntu-2204) instructions. To uninstall them, following what's written [here](https://github.com/abraunegg/onedrive/blob/master/docs/INSTALL.md#uninstalling-the-client), it says to do sudo make uninstall from "within your GitHub repository clone", but.... I do not have a GitHub repository clone?

You missed the whole part where it says if you uninstalled from a 'package' (which is what you did when you followed the Ubuntu instructions, you need to follow the distribution way of uninstalling packages:

![image](https://github.com/abraunegg/onedrive/assets/4956234/7d80175c-03e8-4478-ad8e-e4f8694f391c)

Because you are using Ubuntu, this sould be:
```
sudo apt remove onedrive
```

**Comment by abraunegg** [comment]
@annatomaselli 
> So just to confirm, if I were to manually delete all the downloaded OneDrive folders locally without syncing on the terminal afterwards, that would not affect my original files on the cloud, am I correct??
>And if I were to (understand and) follow the uninstalling instructions mentioned above, would that get rid of all the downloaded files from my machine but not from the OneDrive cloud?

If you were to delete your local files, without syncing, then yes, no change .. *however*:

1. You are using Ubuntu .........
2. You are using a Ubuntu package

Your correct steps should be:

1. Remove the package
2. Reboot your system to ensure that Ubuntu does not have any lurking process in the background
3. Perform local system cleanup

The reason for this is, depending on *how* you installed the package (some folk still just install from the Ubuntu Universe repository which installs an older version) then install the correct version over the top.

The Ubuntu Universe packages contain an issue where the package installs a default systemd service to sync your data in the background ............ so if you do a local delete first .. your data *will* get deleted online.

This issue is specific to *only* Ubuntu and its clones (Linux Mint etc) and does not impact other distributions.

**Reply by annatomaselli** [reply]
Thank you so much for your help, much appreciated!
